### PR TITLE
[Doc] Update dataset_prepare & inference

### PR DESCRIPTION
**Pull request by csatsurnh**
as title

### CODE REVIEW

File: docs/zh_cn/user_guides/3_inference.md [hunk 1]
```diff
@@ -1,3 +1,244 @@
...
+>>> images = [image1, image2, ...] # image1 可以是文件路径或 np.ndarray
+>>> inferencer(images, show=True, wait_time=0.5) # wait_time 是延迟时间，0 表示无限
+
+# 或输入图像目录
+>>> images = $IMAGESDIR
+>>> inferencer(images, show=True, wait_time=0.5)
+
+# 保存可视化渲染色图和预测结果
+# out_dir 是保存输出结果的目录，img_out_dir 和 pred_out_dir 为 out_dir 的子目录
+# 以保存可视化渲染色图和预测结果
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:55]:
```suggestion
# 保存可视化的渲染彩色图和预测结果
# out_dir 是保存输出结果的目录，img_out_dir 和 pred_out_dir 为 out_dir 的子目录
# 以保存可视化的渲染彩色图和预测结果
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 2]
```diff
@@ -1,3 +1,244 @@
...
+>>> inferencer(images, out_dir='outputs', img_out_dir='vis', pred_out_dir='pred')
+```
+
+推理器有一个可选参数 `return_datasamples`，其默认值为 False，推理器的返回值默认为 `字典` 类型，包括 'visualization' 和 'predictions' 两个 key。
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:59]:
```suggestion
推理器有一个可选参数 `return_datasamples`，其默认值为 False，推理器的返回值默认为 `dict` 类型，包括 'visualization' 和 'predictions' 两个 key。
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 3]
```diff
@@ -1,3 +1,244 @@
...
+如果 `return_datasamples=True` 推理器将返回 [`SegDataSample`](../advanced_guides/structures.md) 或其列表。
+
+```
+result = inferencer('demo/demo.png')
+# 结果是一个包含 'visualization' 和 'predictions' 两个 key 的 `字典`
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:64]:
```suggestion
# 结果是一个包含 'visualization' 和 'predictions' 两个 key 的 `dict`
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 4]
```diff
@@ -1,3 +1,244 @@
...
+# 'visualization' 包含颜色分割图
+print(result['visualization'].shape)
+# (512, 683, 3)
+
+# 'predictions' 包含带有标签标记的分割掩膜
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:69]:
```suggestion
# 'predictions' 包含带有标签索引的分割掩膜
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 5]
```diff
@@ -1,3 +1,244 @@
...
+
+results = inferencer(images, return_datasamples=True)
+# <class 'list'>
+print(type(results[0]))
+# <class 'mmseg.structures.seg_data_sample.SegDataSample'>
+```
+
+### 初始化
+
+`MMSegInferencer` 必须从 `model` 初始化，该 `model` 可以是模型名称或一个 `Config`，甚至可以是配置文件的路径。
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:93]:
```suggestion
`MMSegInferencer` 必须使用 `model` 初始化，该 `model` 可以是模型名称或一个 `Config`，甚至可以是配置文件的路径。
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 6]
```diff
@@ -1,3 +1,244 @@
...
+模型名称可以在模型的元文件中找到，比如 maskformer 的一个模型名称是 `maskformer_r50-d32_8xb2-160k_ade20k-512x512`，如果输入模型名称，模型的权重将自动下载。以下是其他输入参数：
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:94]:
```suggestion
模型名称可以在模型的元文件（configs/xxx/metafile.yaml）中找到，比如 maskformer 的一个模型名称是 `maskformer_r50-d32_8xb2-160k_ade20k-512x512`，如果输入模型名称，模型的权重将自动下载。以下是其他输入参数：
```

Comment (xiexinch) [reply docs/zh_cn/user_guides/3_inference.md:94]: Might modify the corresponding English document.

File: docs/zh_cn/user_guides/3_inference.md [hunk 7]
```diff
@@ -1,3 +1,244 @@
...
+
+- weights（str，可选）- 权重的路径。如果未指定，并且模型是元文件中的模型名称，则权重将从元文件加载。默认为 None。
+- classes（list，可选）- 输入类别用于结果渲染，由于分割模型的预测是具有标签索引的分割图，`classes` 是一个响应标签索引的项目列表。若 classes 没有定义，可视化工具将默认使用 `cityscapes` 的类别。默认为 None。
+- palette（list，可选）- 输入配色用于结果渲染，它是响应类的配色列表。若 palette 没有定义，可视化工具将默认使用 `cityscapes` 配色。默认为 None。
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:98]:
```suggestion
- palette（list，可选）- 输入调色盘用于结果渲染，它是对应分类的配色列表。若 palette 没有定义，可视化工具将默认使用 `cityscapes` 的调色盘。默认为 None。
```

File: docs/en/user_guides/3_inference.md [hunk 8]
```diff
@@ -94,16 +93,10 @@ print(type(results[0]))
...
-  responding to the classes. If palette is not defined, visualizer will take `cityscapes` palette by default. Defaults to None.
-- dataset_name (str, optional)[Dataset name or alias](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/utils/class_names.py#L302-L317)
-  visulizer will use the meta information of the dataset i.e. classes and palette,
-  but the `classes` and `palette` have higher priority. Defaults to None.
+- weights (str, optional) -  Path to the checkpoint. If it is not specified and model is a model name of metafile, the weights will be loaded from metafile. Defaults to None.
+- classes (list, optional) - Input classes for result rendering, as the prediction of segmentation model is a segment map with label indices, `classes` is a list which includes items responding to the label indices. If classes is not defined, visualizer will take `cityscapes` classes by default. Defaults to None.
+- palette (list, optional) - Input palette for result rendering, which is a list of color palette responding to the classes. If palette is not defined, visualizer will take `cityscapes` palette by default. Defaults to None.
```
Comment (xiexinch) [inline docs/en/user_guides/3_inference.md:98]:
```suggestion
- palette (list, optional) - Input palette for result rendering, which is a list of colors corresponding to the classes. If the palette is not defined, the visualizer will take the palette of `cityscapes` by default. Defaults to None.
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 9]
```diff
@@ -1,3 +1,244 @@
...
+- dataset_name（str，可选）- [数据集名称或别名](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/utils/class_names.py#L302-L317)，可视化工具将使用数据集的元信息，如类别和配色，但 `classes` 和 `palette` 具有更高的优先级。默认为 None。
+- device（str，可选）- 运行推理的设备。如果无，则会自动使用可用的设备。默认为 None。
+- scope（str，可选）- 模型的作用域。默认为 'mmseg'。
+
+### 可视化预测
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:103]:
```suggestion
### 可视化预测结果
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 10]
```diff
@@ -1,3 +1,244 @@
...
+
+- `SegDataSample` 或 list\[`SegDataSample`\]：如果 imgs 是列表或元组，则返回相同长度的列表类型结果，否则直接返回分割结果。
+
+**注意：** [SegDataSample](https://github.com/open-mmlab/mmsegmentation/blob/1.x/mmseg/structures/seg_data_sample.py) 是 MMSegmentation 的数据结构接口，用作不同组件之间的接口。`SegDataSample` 实现抽象数据元素 `mmengine.structures.BaseDataElement`，请参阅 [MMEngine](https://github.com/open-mmlab/mmengine) 中的数据元素[文档](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/data_element.html)了解更多信息。
+
+`SegDataSample` 中的参数分为几个部分：
+
+- `gt_sem_seg`（`PixelData`）- 语义分割的标注。
+- `pred_sem_seg`（`PixelData`）- 语义分割的预测。
+- `seg_logits`（`PixelData`）- 语义分割的预测指标。
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/3_inference.md:178]:
```suggestion
- `seg_logits`（`PixelData`）- 模型最后一层的输出结果。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 11]
```diff
@@ -1,3 +1,622 @@
-## 准备数据集（待更新）
+# 教程2：准备数据集
 
-中文版文档支持中，请先阅读[英文版本](../../en/user_guides/2_dataset_prepare.md)
+我们建议将数据集根目录符号链接到 `$MMSEGMENTATION/data`。
+如果您的文件夹结构不同，您可能需要更改配置文件中相应的路径。
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:4]:
```suggestion
如果您的目录结构不同，您可能需要更改配置文件中相应的路径。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 12]
```diff
@@ -1,3 +1,622 @@
...
+
+## CHASE DB1
+
+CHASE DB1 的训练和验证集可以从[此处](https://staffnet.kingston.ac.uk/~ku15565/CHASE_DB1/assets/CHASEDB1.zip)下载。
+
+要将 CHASE DB1 数据集转换为 MMSegmentation 的格式，请运行以下命令：
+
+```shell
+python tools/dataset_converters/drive.py /path/to/training.zip /path/to/test.zip
+```
```
Comment (xiexinch) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:277]:
```suggestion
```shell
python tools/dataset_converters/chase_db1.py /path/to/CHASEDB1.zip
```
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 13]
```diff
@@ -1,3 +1,622 @@
...
+
+```shell
+python tools/dataset_converters/chase_db1.py /path/to/CHASEDB1.zip
+```
+
+这个脚本将自动生成目录结构。
+
+## DRIVE
+
+DRIVE 的训练和验证集可从[此处](https://drive.grand-challenge.org/)下载。但在下载之前，需要注册一个帐户。目前官方尚未提供 '1st_manual'。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:283]:
```suggestion
按照[官网](https://drive.grand-challenge.org/)要求，注册并登陆后，便可以下载 DRIVE 的训练和验证数据集。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 14]
```diff
@@ -1,3 +1,622 @@
...
+
+按照惯例，`/path/to/coco_stuff164k/annotations/*2017/*_labelTrainIds.png` 中的 mask 标注用于 COCO Stuff 164k 的训练和测试。
+
+此数据集的详细信息可在[此处](https://github.com/nightrome/cocostuff#downloads)找到。
+
+## CHASE DB1
+
+CHASE DB1 的训练和验证集可以从[此处](https://staffnet.kingston.ac.uk/~ku15565/CHASE_DB1/assets/CHASEDB1.zip)下载。
+
+要将 CHASE DB1 数据集转换为 MMSegmentation 的格式，请运行以下命令：
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:273]:
```suggestion
请运行以下命令，准备 CHASE DB1 数据集：
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 15]
```diff
@@ -1,3 +1,622 @@
...
+
+DRIVE 的训练和验证集可从[此处](https://drive.grand-challenge.org/)下载。但在下载之前，需要注册一个帐户。目前官方尚未提供 '1st_manual'。
+
+要将 DRIVE 数据集转换为 MMSegmentation 的格式，请运行以下命令：
+
+```shell
+python tools/dataset_converters/drive.py /path/to/training.zip /path/to/test.zip
+```
+
+这个脚本将自动生成目录结构。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:291]: same problem as above

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 16]
```diff
@@ -1,3 +1,622 @@
...
+
+## HRF
+
+首先，下载 [health.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/healthy.zip)、[glaucoma.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/glaucoma.zip)、[diabetic_retinopathy.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/diabetic_retinopathy.zip)、[healthy_manualsegm.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/healthy_manualsegm.zip)、[glaucoma_manualsegm.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/glaucoma_manualsegm.zip) 和 [diabetic_retinopathy_manualsegm.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/diabetic_retinopathy_manualsegm.zip)。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:295]:
```suggestion
请下载 [health.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/healthy.zip)、[glaucoma.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/glaucoma.zip)、[diabetic_retinopathy.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/diabetic_retinopathy.zip)、[healthy_manualsegm.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/healthy_manualsegm.zip)、[glaucoma_manualsegm.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/glaucoma_manualsegm.zip) 和 [diabetic_retinopathy_manualsegm.zip](https://www5.cs.fau.de/fileadmin/research/datasets/fundus-images/diabetic_retinopathy_manualsegm.zip)，无需解压，可以直接运行以下命令，准备 HRF 数据集：
```“首先“ 之后一般需要跟 ”然后“，如果没有"然后"，建议删掉“首先”

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 17]
```diff
@@ -1,3 +1,622 @@
...
+
+要将 HRF 数据集转换为 MMSegmentation 的格式，请运行以下命令：
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:297]:
```suggestion
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 18]
```diff
@@ -1,3 +1,622 @@
...
+
+```shell
+python tools/dataset_converters/hrf.py /path/to/healthy.zip /path/to/healthy_manualsegm.zip /path/to/glaucoma.zip /path/to/glaucoma_manualsegm.zip /path/to/diabetic_retinopathy.zip /path/to/diabetic_retinopathy_manualsegm.zip
+```
+
+这个脚本将自动生成目录结构。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:303]: same problem as above

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 19]
```diff
@@ -1,3 +1,622 @@
...
+
+由于我们只支持在此数据集上的模型测试，因此您只需要下载[验证集](https://data.vision.ee.ethz.ch/csakarid/shared/GCMA_UIoU/Dark_Zurich_val_anon.zip)。
+
+## Nighttime Driving
+
+由于我们只支持在此数据集上的模型测试，因此您只需要下载[测试集](http://data.vision.ee.ethz.ch/daid/NighttimeDriving/NighttimeDrivingTest.zip)。
+
+## LoveDA
+
+数据可以从 Google Drive [此处](https://drive.google.com/drive/folders/1ibYV0qwn4yuuh068Rnc-w4tPi0U0c-ti?usp=sharing)下载。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:327]:
```suggestion
数据可以从[此处](https://drive.google.com/drive/folders/1ibYV0qwn4yuuh068Rnc-w4tPi0U0c-ti?usp=sharing)下载 LaveDA 数据集。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 20]
```diff
@@ -1,3 +1,622 @@
...
+```shell
+# 下载 Train.zip
+wget https://zenodo.org/record/5706578/files/Train.zip
+# 下载 Val.zip
+wget https://zenodo.org/record/5706578/files/Val.zip
+# 下载 Test.zip
+wget https://zenodo.org/record/5706578/files/Test.zip
+```
+
+对于 LoveDA 数据集，请运行以下命令重组数据集。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:340]:
```suggestion
对于 LoveDA 数据集，请运行以下命令调整数据集目录。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 21]
```diff
@@ -1,3 +1,622 @@
...
+使用训练好的模型来预测 LoveDA 的测试集并将结果提交给服务器可以在[这里](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/docs/en/user_guides/3_inference.md)找到。
+
+有关 LoveDA 的更多详细信息，可查看[此处](https://github.com/Junjue-Wang/LoveDA).
+
+## ISPRS Potsdam
+
+[Potsdam](https://www2.isprs.org/commissions/comm2/wg4/benchmark/2d-sem-label-potsdam/) 城市语义分割数据集用于 2D 语义分割竞赛 —— Potsdam。
+
+数据集可以在竞赛[主页](https://www2.isprs.org/commissions/comm2/wg4/benchmark/data-request-form/)上请求获得。
+需要 '2_Ortho_RGB.zip' 和 '5_Labels_all_noBoundary.zip'。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:355]:
```suggestion
实验中需要下载 '2_Ortho_RGB.zip' 和 '5_Labels_all_noBoundary.zip'。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 22]
```diff
@@ -1,3 +1,622 @@
...
+数据集可以在竞赛[主页](https://www2.isprs.org/commissions/comm2/wg4/benchmark/data-request-form/)上请求获得。
+需要 'ISPRS_semantic_labeling_Vaihingen.zip' 和 'ISPRS_semantic_labeling_Vaihingen_ground_truth_eroded_COMPLETE.zip'。
+
+对于 Vaihingen 数据集，请运行以下命令重组数据集。
+
+```shell
+python tools/dataset_converters/vaihingen.py /path/to/vaihingen
+```
+
+在我们的默认设置（`clip_size`=512, `stride_size`=256）中，将生成 344 张图像用于训练和 398 张图像用于验证。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:378]: same problem as above

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 23]
```diff
@@ -1,3 +1,622 @@
...
+
+## iSAID
+
+图像数据可从 [DOTA-v1.0](https://captain-whu.github.io/DOTA/dataset.html)（训练/验证/测试）下载
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:382]:
```suggestion
iSAID 数据集可从 [DOTA-v1.0](https://captain-whu.github.io/DOTA/dataset.html) 下载训练/验证/测试数据集的图像数据，
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 24]
```diff
@@ -1,3 +1,622 @@
...
+
+数据标注可以从 [iSAID](https://captain-whu.github.io/iSAID/dataset.html)（训练/验证）下载
+
+该数据集是航空图像实例分割（也有语义分割）的大规模数据集。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:386]:
```suggestion
该数据集是航空图像实例分割和语义分割任务的大规模数据集。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 25]
```diff
@@ -1,3 +1,622 @@
...
+
+```shell
+python tools/dataset_converters/synapse.py --dataset-path /path/to/synapse
+```
+
+注意，MMSegmentation 的默认评估指标（例如 mean dice value）是在 2D 切片图像上计算的，这与 [TransUNet](https://arxiv.org/abs/2102.04306) 等一些论文中的 3D 扫描结果是不同的。
+
+## REFUGE
+
+在 [REFUGE Challenge](https://refuge.grand-challenge.org) 注册并下载 [REFUGE 数据集](https://refuge.grand-challenge.org/REFUGE2Download)。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:535]:
```suggestion
在 [REFUGE Challenge](https://refuge.grand-challenge.org) 官网上注册并下载 [REFUGE 数据集](https://refuge.grand-challenge.org/REFUGE2Download)。
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 26]
```diff
@@ -1,3 +1,622 @@
...
+│   │   │   ├── training
+│   │   │   ├── validation
+│   │   │   ├── test
+```
+
+包含 400 张用于训练的图像、400 张用于验证的图像和 400 张用于测试的图像，这与 REFUGE 2018 数据集相同。
+
+## Mapillary Vistas Datasets
+
+- 注册后，数据集可在[此处](https://www.mapillary.com/dataset/vistas)下载。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/2_dataset_prepare.md:575]: same problem as above

File: docs/zh_cn/user_guides/3_inference.md [hunk 27]
```diff
@@ -1,3 +1,244 @@
...
+>>> inferencer(images, out_dir='outputs', img_out_dir='vis', pred_out_dir='pred')
+```
+
+推理器有一个可选参数 `return_datasamples`，其默认值为 False，推理器的返回值默认为 `dict` 类型，包括 'visualization' 和 'predictions' 两个 key。
+如果 `return_datasamples=True` 推理器将返回 [`SegDataSample`](../advanced_guides/structures.md) 或其列表。
+
+```
+result = inferencer('demo/demo.png')
+# 结果是一个包含 'visualization' 和 'predictions' 两个 key 的 `dict`
+# 'visualization' 包含颜色分割图
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:65]:
```suggestion
# 'visualization' 包含彩色分割图
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 28]
```diff
@@ -1,3 +1,244 @@
...
+>>> inferencer('demo/demo.png', show=True)
+```
+
+可视化结果应如下所示：
+
+<div align="center">
+    <img src='https://user-images.githubusercontent.com/76149310/221507927-ae01e3a7-016f-4425-b966-7b19cbbe494e.png' />
+</div>
+
+此外，您可以使用 `MMSegInferencer` 来处理图像列表：
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:42]:
```suggestion
此外，您可以使用 `MMSegInferencer` 来处理一个包含多张图片的 `list`：
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 29]
```diff
@@ -1,3 +1,244 @@
...
+
+```
+# 输入一个图像列表
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:45]:
```suggestion
# 输入一个图片 list
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 30]
```diff
@@ -1,3 +1,244 @@
...
+>>> images = [image1, image2, ...] # image1 可以是文件路径或 np.ndarray
+>>> inferencer(images, show=True, wait_time=0.5) # wait_time 是延迟时间，0 表示无限
+
+# 或输入图像目录
+>>> images = $IMAGESDIR
+>>> inferencer(images, show=True, wait_time=0.5)
+
+# 保存可视化渲染彩色图和预测结果
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:53]:
```suggestion
# 保存可视化渲染彩色分割图和预测结果
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 31]
```diff
@@ -1,3 +1,244 @@
...
+# out_dir 是保存输出结果的目录，img_out_dir 和 pred_out_dir 为 out_dir 的子目录
+# 以保存可视化渲染彩色图和预测结果
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:55]:
```suggestion
# 以保存可视化渲染彩色分割图和预测结果
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 32]
```diff
@@ -1,3 +1,244 @@
...
+
+# 'predictions' 包含带有标签索引的分割掩膜
+print(result['predictions'].shape)
+# (512, 683)
+
+result = inferencer('demo/demo.png', return_datasamples=True)
+print(type(result))
+# <class 'mmseg.structures.seg_data_sample.SegDataSample'>
+
+# 输入一个图像列表
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:77]:
```suggestion
# 输入一个图片 list
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 33]
```diff
@@ -1,3 +1,244 @@
...
+# <class 'mmseg.structures.seg_data_sample.SegDataSample'>
+```
+
+### 初始化
+
+`MMSegInferencer` 必须使用 `model` 初始化，该 `model` 可以是模型名称或一个 `Config`，甚至可以是配置文件的路径。
+模型名称可以在模型的元文件（configs/xxx/metafile.yaml）中找到，比如 maskformer 的一个模型名称是 `maskformer_r50-d32_8xb2-160k_ade20k-512x512`，如果输入模型名称，模型的权重将自动下载。以下是其他输入参数：
+
+- weights（str，可选）- 权重的路径。如果未指定，并且模型是元文件中的模型名称，则权重将从元文件加载。默认为 None。
+- classes（list，可选）- 输入类别用于结果渲染，由于分割模型的预测是具有标签索引的分割图，`classes` 是一个响应标签索引的项目列表。若 classes 没有定义，可视化工具将默认使用 `cityscapes` 的类别。默认为 None。
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:97]:
```suggestion
- classes（list，可选）- 输入类别用于结果渲染，由于分割模型的预测结构是标签索引的分割图，`classes` 是一个相应的标签索引的类别列表。若 classes 没有定义，可视化工具将默认使用 `cityscapes` 的类别。默认为 None。
```

File: docs/zh_cn/user_guides/3_inference.md [hunk 34]
```diff
@@ -1,3 +1,244 @@
...
+- palette（list，可选）- 输入调色盘用于结果渲染，它是对应分类的配色列表。若 palette 没有定义，可视化工具将默认使用 `cityscapes` 的调色盘。默认为 None。
+- dataset_name（str，可选）- [数据集名称或别名](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/utils/class_names.py#L302-L317)，可视化工具将使用数据集的元信息，如类别和配色，但 `classes` 和 `palette` 具有更高的优先级。默认为 None。
+- device（str，可选）- 运行推理的设备。如果无，则会自动使用可用的设备。默认为 None。
+- scope（str，可选）- 模型的作用域。默认为 'mmseg'。
+
+### 可视化预测结果
+
+`MMSegInferencer` 支持4个用于可视化预测的参数，您可以在调用初始化推理器时使用它们：
```
Comment (MeowZheng) [inline docs/zh_cn/user_guides/3_inference.md:105]:
```suggestion
`MMSegInferencer` 有4个用于可视化预测的参数，您可以在初始化推理器时使用它们：
```

File: docs/zh_cn/user_guides/2_dataset_prepare.md [hunk 35]
```diff
@@ -1,3 +1,622 @@
...
+## Cityscapes
+
+注册后，数据可以在[这里](https://www.cityscapes-dataset.com/downloads/)找到。
+
+按照惯例，`**labelTrainIds.png` 用于 cityscapes 训练。
+我们提供了一个基于 [cityscapesscripts](https://github.com/mcordts/cityscapesScripts) 的[脚本](https://github.com/open-mmlab/mmsegmentation/blob/1.x/tools/dataset_converters/cityscapes.py)用于生成 `**labelTrainIds.png`。
+
+```shell
+# --nproc 表示 8 个转换进程，也可以省略。
+python tools/dataset_converters/cityscapes.py data/cityscapes --nproc 8
```
[the remark this answers is not in the document]
Comment (csatsurnh) [reply docs/zh_cn/user_guides/2_dataset_prepare.md:191]: 好像还是需要用脚本生成 `**labelTrainIds.png`

**Review by xiexinch — APPROVED** [review]

**Pull request merged**